index.js

```
@@ -68,8 +69,7 @@ function TextDecoder(encoding = UTF8, options = {}) {
 }
 
 TextDecoder.prototype.decode = function (buf) {
-  // not sure of if this is possible
-  if (!Buffer.isBuffer(buf) && buf instanceof Uint8Array) {
+  if (!Buffer.isBuffer(buf)) {
```

should we throw on unexpected types?

---

Done!